pkg/collector/collector.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2021 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"github.com/openebs/lvm-localpv/pkg/lvm"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+)
+
+/*
+lvmCollector collects vg total size, vg free size and lv size metrics
+*/
+type lvmCollector struct {
+	vgFreeMetric *prometheus.Desc
+	vgSizeMetric *prometheus.Desc
+	lvSizeMetric *prometheus.Desc
+}
+
+func NewLvmCollector() prometheus.Collector {
+	return &lvmCollector{
+		vgFreeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "free_size_bytes"),
+			"LVM VG free size in bytes",
+			[]string{"name"}, nil,
+		),
+		vgSizeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "total_size_bytes"),
+			"LVM VG total size in bytes",
+			[]string{"name"}, nil,
+		),
+		// Metric name is openebs_size_of_volume which stores the size of lv
+		lvSizeMetric: prometheus.NewDesc(prometheus.BuildFQName("openebs", "size_of", "volume"),
```

Comment:
The unit of metric should also be part of the metric name.

Also, is this the common metric for lvm and device?

Reply:
Yes as per the [prometheus best practices](https://prometheus.io/docs/practices/naming/), the metric name should have prefix of unit in plural form. But existing plugins ([jiva](https://github.com/openebs/monitoring/blob/develop/docs/metrics-jiva.md) or cstor) already using above metrics names, so I think we can maintain a consistency with above.